enhancements/installer/aws-lb-subnet-selection.md

```diff
@@ -0,0 +1,686 @@
+
+A user installed a ROSA HCP cluster with 1 MachinePool in one AZ similar to
+[ROSA HCP: 1 MachinePool in 1 Availability Zone](#rosa-hcp-1-machinepool-in-1-availability-zone),
+but now they want to add another machine pool in a different AZ. The user
+would need to adjust the existing default IngressController subnets by following
+the workflow [Updating an existing IngressController with new Subnets Workflow](/enhancements/ingress/lb-subnet-selection-aws.md#updating-an-existing-ingresscontroller-with-new-subnets-workflow).
+
+Additionally, if the user wants all future IngressController to the new subnet by default, see
+[Updating the Default Subnets for New IngressControllers Workflow](#updating-the-default-subnets-for-new-ingresscontrollers-workflow).
+
+#### Dedicated Ingress Subnets for Security
+
+_"As a cluster admin, I want to install a cluster with dedicated IngressController
+subnets in order to have load balancer VIPs on a dedicated VLAN for applying ACLs."_
+
+A cluster admin wants their load balancers mapped to separate subnet(s), allowing
+firewall ACL rules to specifically target the load balancer VLAN
+(see https://access.redhat.com/support/cases/#/case/03054638). The dedicated subnet(s)
+must be located in the same AZ(s) as the provided cluster subnets; otherwise,
+the IngressController will not function properly.
+
+See [Setting IngressController Subnets during Installation Workflow](#setting-ingresscontroller-subnets-during-installation-workflow)
+for the workflow for this user story.
+
+### Goals
+
+- Deprecate the `platform.aws.subnets` field in the install-config and replace with a
+  more flexible subnet structure for subnet-related configuration.
+- Add a new field in the install-config for IngressController subnet selection for AWS.
+- Provide install-time validation for user provided ingress subnets.
+- Support configuring the default IngressController's subnets at install time for AWS.
+- Support configuring default subnets for all new IngressControllers at install time for AWS.
+- Support updating the default subnets for all new IngressControllers.
+
+### Non-Goals
+
+- Extend support to platforms other than AWS.
+- Automatically restrict IngressController subnet selection for private clusters to ensure the cluster stays private.
+- Assume that bring-your-own (BYO) cluster subnets are also ingress subnets by default.
+- Support configuring subnets for LoadBalancer-type Services that aren't associated with an IngressController.
+
+## Proposal
+
+To enable cluster admins to specify IngressController subnets at install time, we will need
+to make the following API updates:
+
+1. Deprecate the `platform.aws.subnets` field in the install-config and replace with
+   `platform.aws.subnetsConfig`, which more intuitively manages both cluster and ingress subnets selection.
+2. Add the `Subnets` field to `spec.loadBalancer.platform.aws.subnets` in the IngressConfig
+   to encode the default subnets for IngressControllers.
```

Comment:
Do you mean we'll end end with a field called `spec.loadBalancer.platform.aws.subnets.subnets`?